app/portainer/components/form-components/FileUpload/FileUploadField.tsx

```diff
@@ -0,0 +1,70 @@
+import { ChangeEvent, createRef, useState } from 'react';
+
+import { r2a } from '@/react-tools/react2angular';
+import { Button } from '@/portainer/components/Button';
+
+import styles from './FileUploadField.module.css';
+
+export interface Props {
+  onChange(value: unknown): void;
+  title?: string;
+  file?: File;
+  required?: boolean;
+}
+
+export function FileUploadField({
+  required = false,
+  title = 'Select a file',
+  onChange,
+  file,
```

Comment:
let's go with the default form components `file, onChange`

Reply:
not sure I understand this comment, do you mean to change the order of the props?

Reply:
I mean, instead of using `file`, use `value`